topo/node/cptx/cptx.go

```diff
@@ -85,6 +87,64 @@ func (n *Node) SpawnCLIConn() error {
 	return err
 }
 
+// GenerateSelfSigned generates a self-signed TLS certificate using Junos PKI
+func (n *Node) GenerateSelfSigned(ctx context.Context) error {
+	selfSigned := n.Proto.GetConfig().GetCert().GetSelfSigned()
+	if selfSigned == nil {
+		log.Infof("%s - no cert config", n.Name())
+		return nil
+	}
+	log.Infof("%s - generating self signed certs", n.Name())
+	log.Infof("%s - waiting for pod to be running", n.Name())
+	w, err := n.KubeClient.CoreV1().Pods(n.Namespace).Watch(ctx, metav1.ListOptions{
+		FieldSelector: fields.SelectorFromSet(
+			fields.Set{metav1.ObjectNameField: n.Name()},
+		).String(),
+	})
+	if err != nil {
+		return err
+	}
+	for e := range w.ResultChan() {
+		p := e.Object.(*corev1.Pod)
+		if p.Status.Phase == corev1.PodRunning {
+			break
+		}
+	}
+	log.Infof("%s - pod running.", n.Name())
+
+	if err := n.SpawnCLIConn(); err != nil {
+		return err
+	}
+
+	if !n.cliConn.Transport.IsAlive() {
+		return errors.New("scrapligo device driver not open")
+	}
+
+	commands := []string{
+		fmt.Sprintf("request security pki generate-key-pair certificate-id %s", selfSigned.GetCertName()),
+		fmt.Sprintf("request security pki local-certificate generate-self-signed certificate-id %s "+
+			"subject CN=abc domain-name google.com ip-address 1.2.3.4 email example@google.com",
```

Comment:
is this string specific to google's use case? or is this okay for other external users of KNEs?

Reply:
This will work for all users of KNE. With -tls_skip_verify host and ip are not verified hence these can be static values as of now. These are mandatory args.